src/test/java/org/datadog/jmxfetch/TestReconnectContainer.java

```diff
@@ -0,0 +1,187 @@
+
+    private static ImageFromDockerfile img = new ImageFromDockerfile()
+        .withFileFromPath(".", Paths.get("./tools/misbehaving-jmx-server/"));
+
+    @Before
+    public void setup() {
+        this.controlClient = new JMXServerControlClient(cont.getHost(), cont.getMappedPort(controlPort));
+    }
+
+    @Rule
+    public GenericContainer<?> cont = new GenericContainer<>(img)
+        .withExposedPorts(rmiPort, controlPort)
+        .withEnv(Collections.singletonMap("RMI_PORT", "" + rmiPort))
+        .withEnv(Collections.singletonMap("CONTROL_PORT", "" + controlPort))
+        .waitingFor(Wait.forLogMessage(".*IAMREADY.*", 1));
+
+    @Test
+    public void testJMXDirectBasic() throws Exception {
+        // Connect directly via JMXConnector
+        String remoteJmxServiceUrl = String.format(
+            "service:jmx:rmi:///jndi/rmi://%s:%s/jmxrmi",
+            cont.getHost(), cont.getMappedPort(rmiPort)
+        );
+
+        log.info("Connecting to jmx url: {}", remoteJmxServiceUrl);
+        JMXServiceURL jmxUrl = new JMXServiceURL(remoteJmxServiceUrl);
+        JMXConnector conn = JMXConnectorFactory.connect(jmxUrl);
+        MBeanServerConnection mBeanServerConnection = conn.getMBeanServerConnection();
+
+        assertEquals(true, isDomainPresent("Bohnanza", mBeanServerConnection));
+    }
+
+    @Test
+    public void testJMXDirectReconnect() throws Exception {
+        // Connect directly via JMXConnector
+        String remoteJmxServiceUrl = String.format(
+            "service:jmx:rmi:///jndi/rmi://%s:%s/jmxrmi",
+            cont.getHost(), cont.getMappedPort(rmiPort)
+        );
+
+        log.info("Connecting to jmx url: {}", remoteJmxServiceUrl);
+        JMXServiceURL jmxUrl = new JMXServiceURL(remoteJmxServiceUrl);
+        JMXConnector conn = JMXConnectorFactory.connect(jmxUrl);
+        MBeanServerConnection mBeanServerConnection = conn.getMBeanServerConnection();
+
+        assertEquals(true, isDomainPresent("Bohnanza", mBeanServerConnection));
+
+        this.controlClient.jmxCutNetwork();
+
+        assertEquals(false, isDomainPresent("Bohnanza", mBeanServerConnection));
+
+        this.controlClient.jmxRestoreNetwork();
+
+        assertEquals(true, isDomainPresent("Bohnanza", mBeanServerConnection));
+    }
+
+    @Test
+    public void testJMXFetchBasic() throws IOException, InterruptedException {
+        this.initApplication("jmxint_container.yaml", cont.getHost(), cont.getMappedPort(rmiPort));
+
+        this.app.doIteration();
+        List<Map<String, Object>> metrics = ((ConsoleReporter) this.appConfig.getReporter()).getMetrics();
+        assertEquals(1, metrics.size());
+    }
+
+    @Test
+    public void testJMXFetchManyMetrics() throws IOException, InterruptedException {
+        int numBeans = 100;
+        int numAttributesPerBean = 4;
+
+        String testDomain = "test-domain";
+        this.controlClient.createMBeans(testDomain, numBeans);
+        this.initApplication(concatWithNewlines(
```

Comment:
```suggestion
        this.initApplicationWithConfigLines((
```

Just fold the `concatWithNewlines` util function into a specialized `initApplication` call